CMakeLists.txt

```diff
@@ -1,6 +1,14 @@
 # Copyright (c) Microsoft Corporation.
 # Licensed under the MIT License.
 
+# Disable in-source builds to prevent source tree corruption.
+if(" ${CMAKE_SOURCE_DIR}" STREQUAL " ${CMAKE_BINARY_DIR}")
```

Comment:
Why do you have a space at the beginning of each string?

Reply:
Not fully sure. I had found this years ago on the CMake docs site, and this is verbatim what it was. I'll fix it to clean it up a bit.